swap_meet/vendor.py

```diff
@@ -1,2 +1,60 @@
+from .item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory = None ):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, inventory_item):
+        self.inventory.append(inventory_item)
+        return inventory_item
+
+    def remove(self, inventory_item):
+        if inventory_item in self.inventory:
+            self.inventory.remove(inventory_item)
+            return inventory_item
+        else:
+            return False
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items 
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
```

Comment:
Here again, I would tend to reverse the sense of this comparison to check for the desired items NOT being in their respective inventories. Then we could treat this as a guard clause (returning False), and then unindenting the main logic.

And unlike the remove logic above, where we could use the error handling of the list remove method to handle the case where an item is missing, doing the explicit check works better here. If we tried to use error handling, we could end up in a situation where maybe the first remove works, but the second fails. We could write code to put things back the way they were, but instead, checking whether everything we need is in place beforehand is much easier.